### ISSUE TITLE
Cmake Error

### ISSUE THREAD

**Issue by NilsHendrikLukas** [encryptogroup/OTExtension#32]
When running `cmake ..` in the build directory I get the following errors..

`CMake Error at CMakeLists.txt:26 (add_library):
  Target "otextension" links to target "Boost::system" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?

CMake Error at CMakeLists.txt:26 (add_library):
  Target "otextension" links to target "Boost::thread" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?

CMake Error at CMakeLists.txt:26 (add_library):
  Target "otextension" links to target "RELIC::relic" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
`

**Comment by dd23** [comment]
I cannot reproduce this issue, neither on Manjaro nor on Ubuntu 19.04.

Can you try to do a fresh clone of the repository? What system are you using? Is boost installed? If yes, is the version at least 1.66?

**Comment by siaaron045** [comment]
I have this error too.What should i do?

**Comment by Nina-9519** [comment]
I have this error, too

**Comment by dd23** [comment]
Please provide info about your system.
Did you try this with a fresh, unmodified clone of the repository?

**Comment by WeiViming** [comment]
@dd23 I have this error, too. Ubuntu 20.04LTS, boost version is 1.71.

**Comment by dd23** [comment]
**Did you try this with a fresh, unmodified clone of the repository?**

**Comment by WeiViming** [comment]
@dd23 Yes, It is fresh, unmodified clone of the repository. I have see another issue about this error. And add `find_package(Boost 1.66.0 COMPONENTS system thread filesystem REQUIRED)` into `CMakeLists.txt`. But it can not fix the last problem:
```
CMake Error at CMakeLists.txt:26 (add_library):
  Target "otextension" links to target "RELIC::relic" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```

**Comment by dd23** [comment]
Maybe you have an old version of ENCRYPTO_utils installed?
Can you check if cmake finds ENCRYPTO_utils, i.e., if it prints ""Found ENCRYPTO_utils".
What's the content of the extern folder? If there is an ENCRYPTO_utils folder in there - which versionis it?

**Comment by WeiViming** [comment]
@dd23 Sorry, I don't know how to see the version of ENCRYPTO_utils. But it prints "Found ENCRYPTO_utils". It is ok. The problem is wrong about "relic" library..  :'< Maybe I should find the reason about "relic". Thank you for your kindness! :)

**Comment by WeiViming** [comment]
I have solved this problem. The key point is remove the ENCRYPTO_utils from your system, and rebuild this project again. We will get this error if we have install the ENCRYPTO_utils by ourselves.

**Comment by QIANG-crypto-230608** [comment]
> I have solved this problem. The key point is remove the ENCRYPTO_utils from your system, and rebuild this project again. We will get this error if we have install the ENCRYPTO_utils by ourselves.

Hello, did you mean delete the installed ENCRYPT_utils from the system and directly run cmake command in the OTExtension  file?  I remove my installed ENCRYPT_utils, but it shows the following errors:

CMake Error at /usr/local/lib/cmake/ENCRYPTO_utils.bak/ENCRYPTO_utilsTargets.cmake:85 (message):
  The imported target "ENCRYPTO_utils::encrypto_utils" references the file

     "/usr/local/lib/libencrypto_utils.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/local/lib/cmake/ENCRYPTO_utils.bak/ENCRYPTO_utilsTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/local/lib/cmake/ENCRYPTO_utils.bak/ENCRYPTO_utilsConfig.cmake:15 (include)
  CMakeLists.txt:10 (find_package)
-----------------------------------------------------------------------------------------
If I run cmake commond of the OTExtensin after installing ENCRYPTO_utils, it shows the following errors:

CMake Error at src/CMakeLists.txt:42 (target_link_libraries):
  Target "encrypto_utils" links to:

    RELIC::relic

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

-- Generating done (0.0s)
